G115: integer overflow conversion uint8 -> int64

### Summary 
uint8 -> int64 has no overflow
### Steps to reproduce the behavior

```go
package main

import (
	"fmt"
)

func main() {
	str := "A\xFF"
	i := int64(str[0])
	fmt.Printf("%d\n", i)
}
```

### gosec version
81cda2f91fbe

### Go version (output of 'go version')
go version go1.22.6 darwin/arm64

### Operating system / Environment
n/a

### Expected behavior
no complaint

### Actual behavior
complains
G115: integer overflow conversion uint8 -> int64:

```
main.go:9:12: G115: integer overflow conversion uint8 -> int64 (gosec)
	i := int64(str[0])
	          ^
```

@ccojocar I updated the repro, it was simplified too much: this repress
```
main.go:9:12: G115: integer overflow conversion uint8 -> int64 (gosec)
	i := int64(str[0])
	          ^
```
```go
package main

import (
	"fmt"
)

func main() {
	str := "A\xFF"
	i := int64(str[0])
	fmt.Printf("%d\n", i)
}
```

--- a/analyzers/conversion_overflow.go
+++ b/analyzers/conversion_overflow.go
@@ -76,7 +76,7 @@ type integer struct {
 }
 
 func parseIntType(intType string) (integer, error) {
-	re := regexp.MustCompile(`(?P<type>u?int)(?P<size>\d{2})?`)
+	re := regexp.MustCompile(`(?P<type>u?int)(?P<size>\d{1,2})?`)
 	matches := re.FindStringSubmatch(intType)
 	if matches == nil {
 		return integer{}, fmt.Errorf("no integer type match found for %s", intType)

--- a/testutils/g115_samples.go
+++ b/testutils/g115_samples.go
@@ -235,4 +235,34 @@ func main() {
 }
 	`,
 	}, 1, gosec.NewConfig()},
+	{[]string{
+		`
+package main
+
+import (
+        "fmt"
+)
+
+func main() {
+        a := "A\xFF"
+        b := int64(a[0])
+        fmt.Printf("%d\n", b)
+}
+	`,
+	}, 0, gosec.NewConfig()},
+	{[]string{
+		`
+package main
+
+import (
+        "fmt"
+)
+
+func main() {
+        var a uint8 = 13
+        b := int(a)
+        fmt.Printf("%d\n", b)
+}
+	`,
+	}, 0, gosec.NewConfig()},
 }

